src/routers/product.router.ts

```diff
@@ -2,21 +2,45 @@ import express, { Request, Response } from "express";
 import { ProductV2Service } from "../services/productV2.service";
 import { IProduct } from "../services/product.interface";
 import { errorHandling } from "./error-handling";
+import { IPagination } from "./pagination.interface";
 
 export const productRouter: express.Router = express.Router();
 const productV2Service = new ProductV2Service();
 
 productRouter.get("", async (req: Request, res: Response) => {
   try {
     const productList = await productV2Service.getAllProducts();
```

Comment:
to something like this on your service, and implement pagination INSIDE the service, not on the router dude.
getAllProductsPaginated(ipagination obj)

Reply:
ok thnaks!